lib/resque/plugins/status/hash.rb

```diff
@@ -1,20 +1,19 @@
-require 'securerandom'
+require "securerandom"
 
 module Resque
   module Plugins
     module Status
-
       # Resque::Plugins::Status::Hash is a Hash object that has helper methods for dealing with
       # the common status attributes. It also has a number of class methods for
       # creating/updating/retrieving status objects from Redis
       class Hash < ::Hash
-
         # Create a status, generating a new UUID, passing the message to the status
         # Returns the UUID of the new status.
         def self.create(uuid, *messages)
           set(uuid, *messages)
-          redis.zadd(set_key, Time.now.to_i, uuid)
-          redis.zremrangebyscore(set_key, 0, Time.now.to_i - @expire_in) if @expire_in
+          score = time_in_nanoseconds
+          redis.zadd(set_key, score, uuid)
```

Comment:
So we were passing in seconds or milliseconds before and the gem update changed this req?

Reply:
Yes, so before we were passing in seconds, which if two were added at the same time their score ended up being the same.

The problem with this is that when retrieving from this sorted set in Redis, `resque-status` reverses the order of the entries by their score (using Redis' `rrevrange`). However, if the score is the same, it sorts them lexicographically which was causing unit tests to randomly fail (due to the `uuid` being randomly generated).

So I updated it to use the most precise value provided by `Time.now`